src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -202,8 +409,41 @@ public class DependencyGraph
                 logger.LogInformation($"Starting build of graph from ({repoUri}@{commit})");
             }
 
+            HashSet<Build> allContributingBuilds = null;
+            HashSet<DependencyDetail> dependenciesMissingBuilds = null;
+            HashSet<Build> rootNodeBuilds = null;
+            Dictionary<DependencyDetail, Build> dependencyCache =
+                new Dictionary<DependencyDetail, Build>(new DependencyDetailComparer());
+
+            if (options.LookupBuilds)
+            {
+                allContributingBuilds = new HashSet<Build>(new BuildComparer());
+                dependenciesMissingBuilds = new HashSet<DependencyDetail>(new DependencyDetailComparer());
+                rootNodeBuilds = new HashSet<Build>(new BuildComparer());
+
+                // Look up the dependency and get the creating build.
+                var barOnlyRemote = remoteFactory.GetBarOnlyRemote(logger);
+                IEnumerable<Build> potentialRootNodeBuilds = await barOnlyRemote.GetBuildsAsync(repoUri, commit);
+                // Filter by those actually producing the root dependencies, if they were supplied
+                if (rootDependencies != null)
+                {
+                    potentialRootNodeBuilds = potentialRootNodeBuilds.Where(b =>
+                        b.Assets.Any(a => rootDependencies.Any(d => a.Name.Equals(d.Name, StringComparison.OrdinalIgnoreCase) &&
+                                          a.Version == d.Version)));
```

Comment:
Don't we have a comparer for this? Let's us it. :-)

Reply:
We don't, but I'll add one